src/main/java/de/photon/aacadditionpro/events/PlayerAdditionViolationCommandEvent.java

```diff
@@ -1,34 +1,59 @@
 package de.photon.aacadditionpro.events;
 
+import de.photon.aacadditionpro.ServerVersion;
 import de.photon.aacadditionpro.modules.ModuleType;
+import de.photon.aacadditionpro.util.exceptions.UnknownMinecraftVersion;
+import lombok.Getter;
 import org.bukkit.Bukkit;
 import org.bukkit.entity.Player;
 import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
 import org.bukkit.event.HandlerList;
-import org.bukkit.event.player.PlayerEvent;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.function.Consumer;
 
 @SuppressWarnings("unused")
-public class PlayerAdditionViolationCommandEvent extends PlayerEvent implements Cancellable
+public class PlayerAdditionViolationCommandEvent extends Event implements Cancellable
 {
     private static final HandlerList handlers = new HandlerList();
+
+    //Needed for 1.8.8
+    public static HandlerList getHandlerList()
+    {
+        return handlers;
+    }
+
+    @NotNull
+    @Override
+    public HandlerList getHandlers()
+    {
+        return handlers;
+    }
+
     private final ModuleType moduleType;
     private boolean cancelled;
     private String command;
 
-    public PlayerAdditionViolationCommandEvent(final Player player, final String command, final ModuleType moduleType)
+    @Getter
+    protected final Player player;
+
+    // Legacy constructor
+    public PlayerAdditionViolationCommandEvent(final Player player, final String command, final ModuleType moduleType, boolean legacy)
     {
-        super(player);
+        super();
```

Comment:
Why you dont use `super(Bukkit.isPrimaryThread());` here too?

Reply:
That is intended, it is the legacy constructor which doesn’t allow for any arguments. Also, it is faster. But it should be negated, which is the bug.

Reply:
ah